cli/flox-rust-sdk/src/flox.rs

```diff
@@ -62,6 +62,8 @@ pub struct Flox {
 
     /// Feature flags
     pub features: Features,
+
+    pub verbosity: i32,
```

Comment:
Suggestion: Store verbosity on the  FloxBuildMK struct  and store _that_ on Flox or store keep _subsystem verbosity_ on flox and initialize in commands/build.rs with that

Reply:
We only have one way to configure this verbosity (`flox -v`) so I think it makes sense to store on `Flox`. Moving it to `FloxBuildMK` implies we could have a different subsystem verbosity. We don't need that now, so I don't think we need to make that possible. YAGNI

Reply:
We actually have two ways to determine verbosity in flox the other one being RUST_LOG.

I'd just rather keep the verbosity or at least Flox out of the interface for build because its setting is specific to the implementator 

Also the verbosity specifically past the cli layer is subsystem only anyway

Reply:
> We actually have two ways to determine verbosity in flox the other one being RUST_LOG.

Right but we have 1 way to configure this verbosity that we want to pass to the Makefile, and for the time being it is exactly equivalent to `flox -v`

Reply:
> and for the time being it is exactly equivalent to flox -v

At the time that is a bit misleading because it doesn't quite fit with the semantics of v's in the logger. Though that is a matter of configuration.

But its also an intrinsic thing value of the makefile (and if we want to apply it further, of nix commands we call) and at least below the cli does not influence the verbosity of flox logs.
I'd therefore argue that the flox libs don't need the notion of verbosity, and setting up the verbosity of the non flox dependencies that we inject into the library functions by means of implementations on Flox or here the builder impl should also happen on the verbosity aware cli layer.

Reply:
> But its also an intrinsic thing value of the makefile

I don't really understand what this is saying.

I think ultimately you want the verbosity of the build to be configurable separate from the verbosity of the CLI itself e.g. to debug a failed build you probably want to know what's happening in the build, and you probably don't care about what's happening in the CLI.

For now, they aren't separately configurable, so it makes sense to store the verbosity on `Flox` since that verbosity is used by both the build and `pkgdb`. The interfaces aren't set in stone, so it's something we can change when it's a priority.

Reply:
> I'd therefore argue that the flox libs don't need the notion of verbosity, and setting up the verbosity of the non flox dependencies that we inject into the library functions by means of implementations on Flox or here the builder impl should also happen on the verbosity aware cli layer.

I agree in principle, particularly if we're trying to keep a clean flox vs flox-rust-sdk distinction. In practice I think that only slows us down. Merging per @zmitchell 's input

Reply:
What I disagree with is that flox/or verbosity for that matter, is required for building. It is a property of the _means_ that implement building.

I really dont think _implementing_ that slows us down.

* Add a config field (incidentally also adding a verbosity config which is nice)
* Add a verbosity field to `FloxBuildMk`
* change `let builder = FloxBuildMk;` to `let builder = FloxBuildMk::new(config.verbosity)` or `let builder = FloxBuildMk::new(flox_to_mk_verbosity(config.verbosity))`

Reply:
> Right but we have 1 way to configure this verbosity that we want to pass to the Makefile, and for the time being it is exactly equivalent to flox -v

Thanks Matthew - it needs to be easy for the user to get more verbose output out of the tool in all of its various contexts, whether that's building, publishing, installing ... whatever. A user does not want the cognitive burden of having to specify a different way to specify verbosity for the build command vs. any other. This isn't unlike `git` and `nix` which both have several subcommands, each of which accepts multiple `-v` flags although they may have completely distinct implementations in each of the different contexts.